workspaces/cli-shared/src/captures/avro/file-system/ingest-s3-capture-saver.ts

```diff
@@ -0,0 +1,161 @@
+import fs from 'fs-extra';
+import {cli} from "cli-ux";
+import stream, { Readable, Writable } from "stream";
+import { getPathsRelativeToConfig, parseRule } from '@useoptic/cli-config';
+import { IgnoreFileHelper } from '@useoptic/cli-config/build/helpers/ignore-file-interface';
+import path from 'path';
+import { IHttpInteraction } from '@useoptic/domain-types';
+//@ts-ignore
+import oboe from 'oboe';
+import AWS, { Endpoint } from "aws-sdk";
+import { CaptureSaver } from './capture-saver';
+import { Token as ContinuationToken, Object as S3Object } from 'aws-sdk/clients/s3';
+import { Command } from "commander";
+import { IIgnoreRunnablePredicate } from '@useoptic/cli-config/build/helpers/ignore-parser';
+
+export async function ingestS3({ 
+  bucketName, 
+  region, 
+  captureId, 
+  pathPrefix, 
+  endpointOverride
+}: {
+  bucketName: string,
+  region?: string,
+  captureId: string,
+  pathPrefix?: string,
+  endpointOverride?: string
+}) {
+  let { capturesPath: capturesBaseDirectory, opticIgnorePath, configPath } = await getPathsRelativeToConfig();
+
+  let ignoreRules = await new IgnoreFileHelper(opticIgnorePath, configPath).getCurrentIgnoreRules();
```

Comment:
In response to @devdoshi -- @jshearer, there is a better facade for this. 

You got ignore rules correctly on 31, use them like this

```
const {shouldIgnore} = parseIgnore(ignoreRules)
shouldIgnore(method, url) /// true
```

Reply:
Done!